gulpfile.js

```diff
@@ -0,0 +1,41 @@
+const gulp = require("gulp");
+const loadPLugins = require("gulp-load-plugins");
+const mjml = require("mjml");
+const path = require("path");
+const shell = require("shelljs");
+
+const plugin = loadPLugins();
+
+const TYPESCRIPT_TEMPLATES_DIR = "lib/templates/html";
+const MJML_TEMPLATES_DIR = "templates/mjml";
+
+const toMjml = (content, options) => {
+  const name = path.basename(options.sourcePath);
+  return [
+    `// DO NOT EDIT THIS FILE`,
+    `// this file was auto generated from '${name}'`,
+    `export default function(`,
+    `  title: string,`,
+    `  headlineText: string,`,
+    `  senderOrganizationName: string,`,
+    `  senderServiceName: string,`,
+    `  titleText: string,`,
+    `  contentHtml: string,`,
+    `  footerHtml: string`,
+    `): string {`,
+    `  return \``,
+    `${mjml.mjml2html(content, { minify: true }).html}\`;`,
+    `}`,
+    ""
+  ].join("\n");
+};
+
+gulp.task("mjml", () => {
```

Comment:
I would call the task with a name that is independent from the underlying templating system used (we could change in the future how we define/generate templates but the task name should not change). Something like `gulp templates` would do :)